src/Menu/Menu.js

```diff
@@ -15,83 +15,103 @@ const Menu = React.createClass({
 
     /**
-     * If true, the width will automatically be
-     * set according to the items inside the menu
-     * using the proper keyline increment.
+     * If true, the width of the menu will automatically be
+     * set according to the widths of its children,
+     * using proper keyline increments (64px for desktop,
+     * 56px otherwise).
      */
     autoWidth: React.PropTypes.bool,
 
     /**
-     * Children for the Menu. Usually MenuItems.
+     * The content of the menu. This is usually used to pass `MenuItem`
+     * elements.
      */
     children: React.PropTypes.node,
 
     /**
-     * Indicates if the menu should render with compact desktop styles.
+     * If true, the menu item will render with compact desktop styles.
      */
     desktop: React.PropTypes.bool,
 
     /**
-     * Disable the auto focus feature.
+     * If true, the menu will not be auto-focused.
      */
     disableAutoFocus: React.PropTypes.bool,
 
     /**
-     * True if this item should be focused by the keyboard initially.
+     * If true, the menu will be keyboard-focused initially.
      */
     initiallyKeyboardFocused: React.PropTypes.bool,
 
     /**
-     * The style object to use to override underlying list style.
+     * Override the inline-styles of the underlying `List` element.
      */
     listStyle: React.PropTypes.object,
 
     /**
-     * The maxHeight of the menu in pixels. If
-     * specified, the menu will scroll if larger than the maxHeight.
+     * The maximum height of the menu in pixels. If specified,
+     * the menu will be scrollable if it's taller than the provided
+     * height.
      */
     maxHeight: React.PropTypes.number,
 
     /**
-     * If true, the value can be an array and allow the menu to be a multi-select.
+     * If true, `value` must be an array and the menu will support
+     * multiple selections.
      */
     multiple: React.PropTypes.bool,
 
     /**
-     * Fired when a menu item is touchTapped and the menu item
-     * value is not equal to the current menu value.
+     * Callback function fired when a menu item with `value` not
+     * equal to the current `value` of the menu is touch-tapped.
+     *
+     * @param {object} event TouchTap event targeting the menu item.
+     * @param {any}  value If `multiple` is true, the menu's `value`
+     * array with either the menu item's `value` added (if
+     * it wasn't already selected) or omitted (if it was already selected).
+     * Otherwise, the `value` of the menu item.
      */
     onChange: React.PropTypes.func,
 
     /**
-     * Fired when an Esc key is keyed down.
+     * Callback function fired when the Esc key is pressed and the menu
+     * is focused.
+     *
+     * @param {object} event `keydown` event targeting the menu.
      */
     onEscKeyDown: React.PropTypes.func,
 
     /**
-     * Fired when a menu item is touchTapped.
+     * Callback function fired when a menu item is touch-tapped.
```

Comment:
I've actually been using 'touch-tapped' on most of the components because the `onTouchTap` handlers are used to handle both touch and mouse click events, so I thought it was important to distinguish it from just a touch event.